Tools/WebKitTestRunner/skia/TestInvocationSkia.cpp

```diff
@@ -44,12 +44,12 @@ namespace WTR {
 static std::string computeSHA1HashStringForPixmap(const SkPixmap& pixmap)
 {
     size_t pixelsWidth = pixmap.width();
-    size_t pixelsHight = pixmap.height();
+    size_t pixelsHeight = pixmap.height();
     size_t bytesPerRow = pixmap.info().minRowBytes();
 
     SHA1 sha1;
-    const auto* bitmapData = pixmap.addr8();
-    for (size_t row = 0; row < pixelsHight; ++row) {
+    const auto* bitmapData = reinterpret_cast<const uint8_t*>(pixmap.addr32());
```

Comment:
This is only valid if the image uses a 32bpp pixel format, which might not be the case. Instead, I think this should work like this:

- Use `pixmap.addr()` here, which returns a `void*` and does not assert on the pixel storage size.
- Then to `.addBytes()` we should be passing `std::span { bitmapData, pixmap.info().minRowBytes() }`, instead of doing the multiplication by 4 ourselves here.
- Finally, `bitmapData` needs to be increased by `pixman.rowBytes()`. This is because each line may have padding at the end, so `minRowBytes() <= rowBytes()` is always true but they may not match.

Alternatively, if we don't want to do calculations ourserves, we can use `pixmap.info().computeOffset(0, row, pixmap.rowBytes())` to get the base pointer for each row.

Reply:
Maybe it's a bit more intuitive to think in terms of “rowBytes is the same as stride”, and “minRowBytes is related to the pixel width”.

Reply:
Chatting with @carlosgcampos it seems better to audit all our uses of the `addrN()` functions and make a follow-up patch, sio feel free to land this as-is for now.